File: app/src/main/java/com/zjgsu/ai/calibrationtest/Area.java
package com.zjgsu.ai.calibrationtest;

import android.graphics.RectF;

import com.google.gson.Gson;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Double on 2017/4/24.
 */

public class Area {

    private static final String AREA = "area";

    private MyRectF[] rectFs;

    private Gson gson = new Gson();

    public Area(MyRectF[] points) {
        this.rectFs = points;
    }

    public MyRectF[] getRects() {
        return rectFs;
    }

    public Area(JSONObject json) throws JSONException {
        String str = json.getString(AREA);
        rectFs = gson.fromJson(str, MyRectF[].class);
    }

    public JSONObject toJSON() throws JSONException {
        JSONObject json = new JSONObject();
        String str = gson.toJson(rectFs);
        json.put(AREA, str);

        return json;
    }

}

File: app/src/main/java/com/zjgsu/ai/calibrationtest/Area.kt
package com.zjgsu.ai.calibrationtest

import android.graphics.RectF
import com.google.gson.Gson
import org.json.JSONObject

/**
 * Created by Double on 28/07/2017.
 */

class Area private constructor() {
    companion object {
        private val AREA: String = "area"
    }
    private var rectFs: Array<RectF>? = null
    private var gson: Gson? = null
    fun getRects() = rectFs!!

    constructor(json: JSONObject) : this() {
        var str: String = json.getString(AREA)
        gson = Gson()
        rectFs = gson!!.fromJson(str, Array<RectF>::class.java)
    }
    constructor(rects: Array<RectF>) : this() {
        gson = Gson()
        rectFs = rects
    }
    fun toJSON() : JSONObject {
        var json : JSONObject =  JSONObject()
        var str : String = gson!!.toJson(rectFs)
        json.put(AREA, str)
        return json
    }
}
